framework/src/MonteCarloPlayerMT.cpp

```diff
@@ -0,0 +1,101 @@
+#include "MonteCarloPlayerMT.h"
+#include "GameTypes.h"
+#include "RandomPlayer.h"
+
+namespace Player {
+
+MonteCarloPlayerMT::MonteCarloPlayerMT() {
+    for(unsigned int i = 0; i < NUM_THREADS; ++i) {
+        m_threads.emplace_back(&MonteCarloPlayerMT::simulationThread, this);
+    }
+}
+
+MonteCarloPlayerMT::~MonteCarloPlayerMT() {
+    for(auto& thread : m_threads) thread.~thread();
```

Comment:
I believe this is causing a "terminate called without an active exception" causing the program to crash at the end. From my understanding, the proper way to exit an std::thread is by using thread.join() instead of calling the destructor.

Reply:
yeah this one was tricky. I have the worker threads running an infinite loop to keep from creating a new set of threads each time the player is asked to pick a move. I have updated this to use a boolean flag for the threads to determine when to return instead of the destructor. look carefully through it and make sure I have done it right.

Reply:
Looks good. I tested it and I am no longer getting a warning from the simulation.